isisd: 'tiebreaker' command line funtionality is inconsistent with its implementation

baozhen-H3C opened the pull request:
isisd: The command fast-reroute lfa tiebreaker [downstream | lowest-backup-metric | node-protecting] index (1-255) [level-1 | level-2] will overwrite configurations with the same index but different types. This is because the index is set as the key in frr-isisd.yang. However, the lfa_tiebreaker_cmp function uses a tuple (index, type) as the key. Therefore, the yang file should be modified to stay in sync with the business logic.

Test Scenario:
On RouterA, first configure fast-reroute lfa tiebreaker downstream index 100 level-1, then configure fast-reroute lfa tiebreaker lowest-backup-metric index 100 level-1, and check the configuration:
!
router isis 10
 fast-reroute lfa tiebreaker lowest-backup-metric index 100 level-1
exit
!

Signed-off-by: baozhen-H3C <bao.zhen@h3c.com>

donaldsharp commented:
1) Apply frrbot style suggestions
2) Commit comment should be a `short subject line` and then a description of what really is going on in the commit later, not one big line on the subject.

baozhen-H3C commented:
I have already modified the Commit comment, but my CI seems to encounter a problem:
  AssertionError: Testcase test_ospfv3_hello_tc10_p0 [22]: Failed
  Error: [DUT: r0] missing OSPF neighbor r1 with router-id 100.1.1.1
  assert '[DUT: r0] missing OSPF neighbor r1 with router-id 100.1.1.1' is True
  E AssertionError: Testcase test_ospfv3_hello_tc10_p0 [22]: Failed
  Error: [DUT: r0] missing OSPF neighbor r1 with router-id 100.1.1.1
  assert '[DUT: r0] missing OSPF neighbor r1 with router-id 100.1.1.1' is True

I modified the ISIS code, and this OSPF unit test does not involve ISIS configuration. Therefore, is this CI error a false alarm? Could you please trigger the CI again? Thank you.

donaldsharp commented:
please apply the frrbot style suggestions

Jafaral reviewed (changes requested):
Can you please change the PR/commit title to be a short sentence? move any details to the body of the commit.

riw777 reviewed (commented):
code looks okay ... waiting on title change, etc.

baozhen-H3C commented:
Sorry,  I have changed it. Is this OK?

riw777 commented:
We need to get these done --

https://github.com/FRRouting/frr/pull/16593/checks?check_run_id=29461002987

once those are done I'll work on figuring out what is happening with the ci failure ...

riw777 commented:
@Jafaral can you take another look at this?

riw777 commented:
@Jafaral can you take a look at this to see if we can clear the blocker?

riw777 reviewed (approved):
lgtm

riw777 commented:
@Mergifyio rebase
